src/Indice.Features.Cases.AspNetCore/Services/MyCaseService.cs

```diff
@@ -257,6 +258,16 @@ internal class MyCaseService : BaseCaseService, IMyCaseService
             return caseTypes.ToResultSet();
         }
 
+        public async Task MarkPdfAsRead(ClaimsPrincipal user, Guid caseId) {
+            var userId = user.FindSubjectId();
+            var @case = await GetDbCaseById(caseId, userId);
+            if (@case is null) {
+                throw new CaseNotFoundException("Case not found.");
+            }
+            @case.IsRead = true;
```

Comment:
αν `!@case.Read` τότε κάνε το update. 

---

έχουμε `AsNoTracking` στο query της μεθόδου `GetDbCaseById`, οποτε επειδη δεν κανει track το entity το efcore θα κανει καθε φορα την εξτρα κλήση για το update, ακόμα κι αν είναι ήδη `true`.

> By default, queries that return entity types are tracking. Which means you can make changes to those entity instances and have those changes persisted by SaveChanges()

>No tracking queries are useful when the results are used in a read-only scenario. They're quicker to execute because there's no need to set up the change tracking information. If you don't need to update the entities retrieved from the database, then a no-tracking query should be used

reference https://learn.microsoft.com/en-us/ef/core/querying/tracking